soci/soci_index.go

```diff
@@ -57,26 +56,47 @@ var (
 	errNotLayerType = errors.New("not a layer mediaType")
 )
 
-// nolint:revive
-type SociIndex struct {
-	MediaType    string `json:"mediaType"`
-	ArtifactType string `json:"artifactType"`
-	// descriptors of ztocs
-	Blobs []ocispec.Descriptor `json:"blobs,omitempty"`
-	// descriptor of image manifest
-	Subject ocispec.Descriptor `json:"subject,omitempty"`
+/// Index represents an interface to interact with ORAS/OCI Ref Types.
+/// In the fullness of time, ORAS support will be removed from the snapshotter
+/// and this interface will not be needed.
+type Index interface {
+	/// Returns the media type of the artifact manifest.
+	GetMediaType() string
```

Comment:
nit: this suggestion is idiomatic go
```suggestion
	MediaType() string
```
This applies to all the getters here.
https://go.dev/doc/effective_go#Getters

Reply:
The reason I chose `GetX` is because having a field and method with the same name causes a conflict. https://go.dev/play/p/g82cyp7RcXu

Reply:
Is there a reason to export the fields if we have getters?

Reply:
JSON marshaling won't work with unexported fields. There are ways around it though. (For e.g, [this](https://stackoverflow.com/questions/11126793/json-and-dealing-with-unexported-fields))

Reply:
If exposing the fields is fine, do we need getters?

Reply:
The main reason for having an interface with getters instead of a concrete type is because ORAS and ORT have different names for the field that points to the manifest it's referring. It's called `subject` in ORAS, and `refers` in ORT.  

I could do something like:

```
struct Index {
    ...
    // Deserialize into whichever field that manifest supports
    Refers `json:"refers,omitempty"` // For ORT
    Subject `json:"subject, omitempty"` // For ORAS
    ...
}
```
but I'm not a fan of this solution.

Reply:
How about using ORT throughout the snapshotter and only converting them to ORAS around registry interactions?

Reply:
I've updated it so that `Index` is no longer an interface. Instead, it is now a `struct` with both `Refers` and `Subject` field. During deserialization, the appropriate field is selected. 

I chose this method because: 
- It is the path of least resistance when we eventually phase out ORAS support, since it will only require removing a single field from the `Index` struct. 
- The snapshotter doesn't need to Subject/Refers field when running the container. The only time we utilize the field is when creating the index during ztoc creation.